File: Intermediate/18.Write a program to implement string compression (e.g., aaabb → a3b2)/Solution.java
import java.util.Scanner;

public class StringCompression {
    public static String compressString(String str) {
        StringBuilder compressed = new StringBuilder();
        int count = 1;

        for (int i = 0; i < str.length(); i++) {
            if (i < str.length() - 1 && str.charAt(i) == str.charAt(i + 1)) {
                count++;
            } else {
                compressed.append(str.charAt(i)).append(count);
                count = 1;
            }
        }

        return compressed.toString();
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter a string: ");
        String str = scanner.nextLine();

        String result = compressString(str);
        System.out.println("Compressed string: " + result);
    }
}

File: Intermediate/18.Write a program to implement string compression (e.g., aaabb → a3b2)/Solution.c
#include <stdio.h>
#include <string.h>

void compressString(char str[]) {
    int n = strlen(str);
    int count = 1;

    for (int i = 0; i < n; i++) {
        if (i < n - 1 && str[i] == str[i + 1]) {
            count++;
        } else {
            printf("%c%d", str[i], count);
            count = 1;
        }
    }
    printf("\n");
}

int main() {
    char str[100];
    printf("Enter a string: ");
    scanf("%s", str);

    printf("Compressed string: ");
    compressString(str);

    return 0;
}
